askama_shared/src/generator.rs

```diff
@@ -1106,9 +1129,64 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
                 self.visit_method_call(buf, obj, method, args)?
             }
             Expr::RustMacro(name, args) => self.visit_rust_macro(buf, name, args),
+            Expr::Let(ref variant, ref params, ref val) => self.visit_let(buf, variant, params, val)?,
         })
     }
 
+    fn visit_let(
+        &mut self,
+        buf: &mut Buffer,
+        variant: &MatchVariant<'a>,
+        params: &Option<MatchParameters<'a>>,
```

Comment:
```suggestion
        params: Option<&MatchParameters<'a>>,
```

Please use `Option<&T>` instead of `&Option<T>` :)

Reply:
Yes, that's cleaner. One problem is, that I had to add explicit lifetimes to every &Expr.

Reply:
I think you can have it be `Option<&MatchParameters<'a>>`, if you remove `'a` from `params` in `visit_match_params()`. At least I don't see why it's required there, though I'm only giving it a quick glance. Overall I think most of the `&'a T` can be changed to `&T` :)